src/models/Room.ts

```diff
@@ -0,0 +1,135 @@
+	 */
+	Public = "public",
+	/**
+	 * This visibility indicates that the room will not be included in published room list.
+	 */
+	Private = "private",
+}
+
+export interface RoomCreateOptions {
+	/**
+	 * Extra keys, such as m.federate, to be added to the content of the m.room.create event.
+	 * The server will overwrite the following keys: `creator`, `room_version`.
+	 * Future versions of the specification may allow the server to overwrite other keys.
+	 */
+	creation_content?: {
+		"m.federate"?: boolean,
+		[key: string]: unknown
+	},
+	/**
+	 * A list of state events to set in the new room.
+	 * This allows the user to override the default state events set in the new room.
+	 * The expected format of the state events are an object with `type`, `state_key` and `content` keys set.
+	 * Takes precedence over events set by `preset`, but gets overridden by `name` and `topic` keys.
+	 */
+	initial_state?: {
+		/**
+		 * The content of the event.
+		 */
+		content: any,
+		/**
+		 * The state_key of the state event. Defaults to an empty string.
+		 */
+		state_key?: string,
+		/**
+		 * The type of event to send.
+		 */
+		type: string
+	}[],
+	/**
+	 * A list of user IDs to invite to the room. This will tell the server to invite everyone in the list to the newly created room.
+	 */
+	invite?: string[],
+	invite_3pid?: {
+		/**
+		 * The invitee’s third party identifier.
+		 */
+		address: string;
+		/**
+		 * An access token previously registered with the identity server. Servers can treat this as optional to distinguish between r0.5-compatible clients and this specification version.
+		 */
+		id_access_token: string;
+		/**
+		 * The hostname+port of the identity server which should be used for third party identifier lookups.
+		 */
+		id_server: string;
+		/**
+		 * The kind of address being passed in the address field, for example `email`.
+		 */
+		medium: string;
+	}[],
+	/**
+	 * This flag makes the server set the `is_direct` flag on the `m.room.member` events sent to the users in `invite` and `invite_3pid`.
+	 */
+	is_direct?: boolean;
+	/**
+	 * If this is included, an `m.room.name` event will be sent into the room to indicate the name of the room.
+	 */
+	name?: string;
+	/**
+	 * The power level content to override in the default power level event.
+	 * This object is applied on top of the generated `m.room.power_levels` event content prior to it being sent to the room.
+	 * Defaults to overriding nothing.
+	 */
+	power_level_content_override?: PowerLevelsEventContent,
+	/**
+	 * Convenience parameter for setting various default state events based on a preset.
+	 * 
+	 * If unspecified, the server should use the `visibility` to determine which preset to use.
+	 * A visbility of `public` equates to a preset of `public_chat` and `private` visibility equates to a preset of `private_chat`.
+	 */
+	// Note: Allowing string values for backwards compatibility
+	preset?: RoomPreset|"public_chat"|"private_chat"|"trusted_private_chat",
```

Comment:
For backwards compat, I've allowed the string editions of the enums. It's a bit gross though, so maybe we should just decide whether we like enums at all and just go for it.

Reply:
per other comments, no enums here please :)